ExtraEnemyCustomization/Customizations/Abilities/AttackCustoms/AttackCustomBase.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Agents;
+using EECustom.Events;
+using Enemies;
+using Player;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace EECustom.Customizations.Abilities
+{
+    public abstract class AttackCustomBase<T> : EnemyCustomBase where T : new()
+    {
+        public T MeleeData { get; set; } = new();
+        public T TentacleData { get; set; } = new();
+
+        public override void OnConfigLoaded()
+        {
+            LocalPlayerDamageEvents.MeleeDamage += OnMelee;
+            LocalPlayerDamageEvents.TentacleDamage += OnTentacle;
+        }
+
+        public override void OnConfigUnloaded()
+        {
+            LocalPlayerDamageEvents.MeleeDamage -= OnMelee;
+            LocalPlayerDamageEvents.TentacleDamage -= OnTentacle;
+        }
+
+        private void OnMelee(PlayerAgent player, Agent inflictor, float damage)
+        {
+            Do(player, inflictor, MeleeData);
+        }
+
+        private void OnTentacle(PlayerAgent player, Agent inflictor, float damage)
+        {
+            Do(player, inflictor, TentacleData);
+        }
+
+        private void Do(PlayerAgent player, Agent inflictor, T setting)
+        {
+            if (inflictor.TryCastToEnemyAgent(out var enemy))
```

Comment:
Out parameter might not work? Or is this an extension method?

Reply:
extension, https://github.com/GTFO-Modding/ExtraEnemyCustomization/blob/ae1183a043b6c67d80f0a9c781e4d8518c747f2c/ExtraEnemyCustomization/Extensions/AgentExtension.cs#L12